### PR TITLE
Implemented Authentication

### PR DESCRIPTION
**Pull request by PalmEmanuel**
Solves #33 by adding new commands and full support for custom authentication. Removes our soft dependency on the Az CLI and the Az PowerShell Module, but keeps the functionality the same if they're available.

Strongly related to https://github.com/PSBicep/PSBicep/issues/258.

New commands:
- `Set-BicepNetCredential` - Sets a credential for the current PowerShell session.
  - Without parameters or with `-TenantId` - Allows the user to interactively log in through the browser.
  - With `-AccessToken` - Allows the user to provide an existing token for authentication
- `Clear-BicepNetCredential` - Removes the credential set by `Set-BicepNetCredential`.
- `Get-BicepNetAccessToken` - Gets the access token from the credential set by `Set-BicepNetCredential`

Still missing some verbosity and logging but I thought I'd open the PR, suggestions or additions appreciated!

### CODE REVIEW

File: BicepNet.Core/BicepWrapper.cs [hunk 1]
```diff
@@ -82,6 +87,78 @@ public static void Initialize(ILogger bicepLogger)
         logger = bicepLogger;
     }
 
+    public static void SetAuthentication(string? token = null, string? tenantId = null)
```
Comment (SimonWahlin) [inline BicepNet.Core/BicepWrapper.cs:90]: I think we should put all auth logic in BicepWrapper.Authentication.cs

**Review by SimonWahlin — CHANGES_REQUESTED** [review]
Have a few comments, oterwise it looks good!

File: BicepNet.Core/BicepWrapper.cs (hunk 1 above)
Comment (PalmEmanuel) [reply BicepNet.Core/BicepWrapper.cs]: Good call, I had trouble writing it in a way I found clean before. Pushed some changes that look much better to me, lemme know what you think!

**Review by SimonWahlin — CHANGES_REQUESTED** [review]
Looks good!
Will you fix the merge conflict so we can merge?

**Review by SimonWahlin — APPROVED** [review]

**Pull request merged**